src/geo/LatLng.js

```diff
@@ -53,30 +76,13 @@ L.LatLng.prototype = {
 };
 
 
-// constructs LatLng with different signatures
-// (LatLng) or ([Number, Number]) or (Number, Number) or (Object)
-
 L.latLng = function (a, b, c) {
-	if (a instanceof L.LatLng) {
+	if (a instanceof L.LatLng || a === null) {
 		return a;
 	}
-	if (L.Util.isArray(a) && typeof a[0] !== 'object') {
-		if (a.length === 3) {
-			return new L.LatLng(a[0], a[1], a[2]);
-		}
-		if (a.length === 2) {
-			return new L.LatLng(a[0], a[1]);
-		}
-		return null;
-	}
-	if (a === undefined || a === null) {
-		return a;
-	}
-	if (typeof a === 'object' && 'lat' in a) {
-		return new L.LatLng(a.lat, 'lng' in a ? a.lng : a.lon, a.alt);
-	}
-	if (b === undefined) {
+	try {
+		return new L.LatLng(a, b, c);
+	} catch(err) {

```

Comment:
Again, try/catch here is very harmful. To see the effect of this, try doing benchmarks of old vs new code on jsperf.com and you'll see why I am concerned. It's also generally a good idea to benchmark sensitive code like LatLng construction when doing any changes.